### PR TITLE
Add colors to airflow config command

### PR DESCRIPTION
**Pull request by mik-laj**
I didn't write automatic tests because it is a system-specific thing and we can't write reliable automated tests. Argument parsing is covered by current tests.

I did manual tests.
When I run the `airflow config` command, I see the screen below.
Screenshot 2020-04-16 at 15 19 46
When I run the `airflow config | cat` command, I see the scrceen below.
Screenshot 2020-04-16 at 15 27 45

---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/cli/cli_parser.py [hunk 1]
```diff
@@ -153,6 +153,12 @@ def __init__(self, flags=None, help=None, action=None, default=None, nargs=None,
     ),
     choices=tabulate_formats,
     default="fancy_grid")
+ARG_COLOR = Arg(
+    ('--color',),
+    help="do emit colored output (default: auto-detect)")
+ARG_NO_COLOR = Arg(
+    ('--no-color',),
```
Comment (ashb) [inline airflow/cli/cli_parser.py]: I think it's possible to do this in ArgParse without a separate option. One moment.

Comment (mik-laj) [reply airflow/cli/cli_parser.py]: Oh yes. `choices`. I will update PR.

File: airflow/utils/cli.py [hunk 2]
```diff
@@ -234,3 +234,39 @@ def sigquit_handler(sig, frame):  # pylint: disable=unused-argument
...
+    if sys.platform == 'win32':
+        return False
+    if not hasattr(sys.stdout, 'isatty'):
+        return False
+    if not sys.stdout.isatty():
+        return False
+    if 'COLORTERM' in os.environ:
+        return True
+    term = os.environ.get('TERM', 'dumb').lower()
+    if term in ('xterm', 'linux') or 'color' in term:
```
Comment (ashb) [inline airflow/utils/cli.py:252]: _Strictly_ speaking `TERM=xterm` doesn't support color. This is probably fine :)

File: airflow/cli/cli_parser.py (hunk 1 above)
Comment (ashb) [reply airflow/cli/cli_parser.py]: Nope, I'm making that up.

File: airflow/cli/cli_parser.py [hunk 3]
```diff
@@ -153,6 +153,12 @@ def __init__(self, flags=None, help=None, action=None, default=None, nargs=None,
...
+    help="do not emit colored output (default: auto-detect)")
```
Comment (ashb) [inline airflow/cli/cli_parser.py]:
```suggestion
ARG_COLOR = Arg(
    ('--color',),
    help="do emit colored output (default: auto-detect)",
    action="store_true")
ARG_NO_COLOR = Arg(
    ('--no-color',),
    help="do not emit colored output (default: auto-detect)",
    action="store_false",
    dest="color")
```

And then should_use_colors is a bit simpler.

File: airflow/utils/cli.py [hunk 4]
```diff
@@ -234,3 +234,39 @@ def sigquit_handler(sig, frame):  # pylint: disable=unused-argument
...
+        return True
+    return False
+
+
+def should_use_colors(args) -> bool:
+    """
+    Processes arguments and decides whether to enable color in output
+    """
+    if args.color and args.no_color:
```
Comment (ashb) [inline airflow/utils/cli.py]: https://docs.python.org/3/library/argparse.html#mutual-exclusion

**Comment by ashb** [comment]
Ah that's better, and more "gnu"-like for `ls --color=auto` etc. Nice.

**Review by ashb — APPROVED** [review]

File: airflow/utils/cli.py (hunk 2 above)
Comment (mik-laj) [reply airflow/utils/cli.py:252]: Breeze uses ``xterm``. I tested on Mac OS and Breeze.

**Review by potiuk — APPROVED** [review]

**Pull request merged**